compiler/AST/primitive.cpp

```diff
@@ -702,11 +702,12 @@ initPrimitive() {
   prim_def(PRIM_REQUIRE, "require", returnInfoVoid, false, false);
 }
 
-Map<const char*, VarSymbol*> memDescsMap;
+static Map<const char*, VarSymbol*> memDescsMap;
```

Comment:
I think I'd expect you to remove memDescMap and replace newMemDesc with something like this:

```
 VarSymbol* newMemDesc(const char* str) {
  newMemDesc(new_StringSymbol(str));
}
```

Is there some reason that can't work?